### PR TITLE
[DUOS-2932][risk=no] Use V4 algorithm for all matches

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2932

### Summary
* Remove usages of commercial use
* Replace with non profit use where appropriate
* Use v4 algorithm for all matching efforts
* Significant refactor of the `DataUse` class (and `DataUseBuilder`) to match the [v4 algorithm](https://github.com/DataBiosphere/consent-ontology/blob/develop/docs/Algorithms.md#version-4)

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/models/DataUse.java [hunk 1]
```diff
@@ -1,79 +1,44 @@
...
 import com.google.gson.GsonBuilder;
 import java.util.List;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
 
-@SuppressWarnings({"unused", "SameParameterValue", "WeakerAccess"})
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
 public class DataUse {
```
Comment (okotsopoulos) [inline src/main/java/org/broadinstitute/consent/http/models/DataUse.java:10]: [General Consent question]

Does Consent have the capability to auto-generate these models as part of your API spec definition?

**Review by okotsopoulos — APPROVED** [review]

File: src/main/java/org/broadinstitute/consent/http/models/DataUse.java (hunk 1 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/models/DataUse.java:10]: Actually, no, it doesn't, which reminds me that I need to update the swagger models to match this, thanks for the reminder!

Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/models/DataUse.java:10]: Turns out that `DataUse.yaml` is not affected by these changes, but does need to be rewritten. I'll tackle that separately.

**Review by aarohinadkarni — APPROVED** [review]
looks good 👍

**Pull request merged**